### PR TITLE
dynamo.export uses core_aten_decomp by default in aten mode

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #93133

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: torch/_dynamo/eval_frame.py [hunk 1]
```diff
@@ -530,6 +531,10 @@ def export(
         assert (
             aten_graph
         ), "Specifying a decomposition_table table or tracing mode is illegal without setting aten_graph=True"
+
+    if aten_graph and decomposition_table is None:
+        decomposition_table = core_aten_decompositions
```
Comment (ezyang) [inline torch/_dynamo/eval_frame.py:536]: one problem with setting up the api this way is that the user needs to know that if they want to just make a slight tweak to the default decomps, they have to know to clone core aten decomps first and then twiddle it

Comment (SherlockNoMad) [reply torch/_dynamo/eval_frame.py:536]: How about `get_core_aten_decompositions()` to always return a cloned decomp_table?

Comment (ezyang) [reply torch/_dynamo/eval_frame.py:536]: Well, I suppose that is some helpful QOL (but I wouldn't call it `get_` anymore), but the documentation point (at least) still stands

**Comment by SherlockNoMad** [comment]
Aborted. Moved to https://github.com/pytorch/pytorch/pull/93868

**Pull request closed without merging**